### PR TITLE
Mistral v0.8 Workflow Upgrade Tool

### PR DESCRIPTION
**Pull request by jfryman**
This is an upgrade tool that parses 0.7 workflows and spits out 0.8 compatible workflows. This is for the users out there that we may not know about (HI! :wave:) that may be using Mistral, giving them a painless way to upgrade when the time comes.

### CODE REVIEW

File: tools/upgrade-mistral-wf-0.8 [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+## Mistral YAQL Migration tool
+#
+# This tool inspects a given Mistral workflow, parses it using a few
+# regex filters, and attempts to rewrite the file to adhere to the
+# updated YAQL syntax introduced in https://review.openstack.org/#/c/157741/
+#
+# There appear to be two different ways that the old style YAQL is
+# represented in workflows. This includes:
+#   * bare         - (node: $.node)
+#   * interpolated - (node: 'Node: {$.node}')
```
Comment (dzimine) [inline tools/upgrade-mistral-wf-0.8]: pls be sure to catch { 1 + $.node } expressions.

File: tools/upgrade-mistral-wf-0.8 [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+#   * functions    - (node: range(0, $.nodes).list() )
+#
+# This command runs sed with two passes attempting to clean up a given
+# workflow. This command will also ensure a backup exists in the event
+# that things do not go as expected to prevent data loss.
+#######################################################################
+
+function usage {
+  echo "Usage: `basename $0` <workflow file>"
```
Comment (dzimine) [inline tools/upgrade-mistral-wf-0.8]: Technically it's 'mistral file', workflows, workbooks, actions...
say workflow/workbook file?

File: tools/upgrade-mistral-wf-0.8 (hunk 1 above)
Comment (jfryman) [inline tools/upgrade-mistral-wf-0.8]: Added https://github.com/StackStorm/st2/pull/1131/files#diff-57b2224ac653d88d3d86728c4be4c945R70

**Comment by jfryman** [comment]
Alright, this should be GTG for release. /cc @dzimine @lakshmi-kannan @DoriftoShoes

**Pull request merged**